packages/flutter/lib/src/material/reorderable_list.dart

```diff
@@ -0,0 +1,473 @@
+      _entranceController.forward(from: 0.0);
+    }
+  }
+
+  // Requests animation to the latest next index if it changes during an animation.
+  void _onEntranceStatusChanged(AnimationStatus status) {
+    if (status == AnimationStatus.completed) {
+      setState(() {
+        _requestAnimationToNextIndex();
+      });
+    }
+  }
+
+  // Scrolls to a target context if that context is not on the screen.
+  void _scrollTo(BuildContext context) {
+    if (_scrolling) 
+      return;
+    final RenderObject contextObject = context.findRenderObject();
+    final RenderAbstractViewport viewport = RenderAbstractViewport.of(contextObject);
+    assert(viewport != null);
+    // If and only if the current scroll offset falls in-between the offsets
+    // necessary to reveal the selected context at the top or bottom of the
+    // screen, then it is already on-screen.
+    final double margin = _dropAreaExtent;
+    final double scrollOffset = _scrollController.offset;
+    final double topOffset = max(
+      _scrollController.position.minScrollExtent, 
+      viewport.getOffsetToReveal(contextObject, 0.0).offset - margin,
+    );
+    final double bottomOffset = min(
+      _scrollController.position.maxScrollExtent,
+      viewport.getOffsetToReveal(contextObject, 1.0).offset + margin,
+    );
+    final bool onScreen = scrollOffset <= topOffset && scrollOffset >= bottomOffset;
+    // If the context is off screen, then we request a scroll to make it visible.
+    if (!onScreen) {
+      _scrolling = true;
+      _scrollController.position.animateTo(
+        scrollOffset < bottomOffset ? bottomOffset : topOffset, 
+        duration: _kScrollAnimationDuration, 
+        curve: Curves.easeInOut,
+      ).then((Null none) {
+        setState(() {
+          _scrolling = false;
+        });
+      });
+    }
+  }
+
+  // Wraps children in Row or Column, so that the children flow in
+  // the widget's scrollDirection.
+  Widget _buildContainerForScrollDirection({List<Widget> children}) {
+    if (widget.scrollDirection == Axis.horizontal) {
+      return new Row(children: children);
+    } 
+    return new Column(children: children);
+  }
+
+  // Wraps one of the widget's children in a DragTarget and Draggable.
+  // Handles up the logic for dragging and reordering items in the list.
+  Widget _wrap(Widget toWrap, int index, BoxConstraints constraints) {
+    assert(toWrap.key != null);
+
+    // Starts dragging toWrap.
+    void onDragStarted() {
+      setState(() {
+        _dragging = toWrap.key;
+        _dragStartIndex = index;
+        _ghostIndex = index;
+        _currentIndex = index;
+        _entranceController.forward(from: 1.0);
+      });
+    }
+
+    // Drops toWrap into the last position it was hovering over.
+    void onDragEnded() {
+      setState(() {
+        if (_dragStartIndex != _currentIndex)
+          widget.onSwap(_dragStartIndex, _currentIndex);
+        _ghostController.reverse(from: 0.1);
```

Comment:
A comment that explained why we're animating from 0.1 would be helpful here.

Reply:
Done. I put this together before I had the close animation spec. I'll see if I can bring this in line with that spec easily.